src/react/DeviceStatusPage/DeviceLiveEventsCard.tsx

```diff
@@ -0,0 +1,347 @@
+          <Badge variant="secondary" className="mr-2">
+            channel
+          </Badge>
+          <span className="text-secondary">{event.message}</span>
+        </li>
+      );
+  }
+};
+
+interface EventDelegateProps {
+  event: RenderableEvent;
+}
+
+const EventDelegate = ({ event }: EventDelegateProps) => {
+  if (isSystemEvent(event)) {
+    return <SystemEventDelegate event={event} />;
+  }
+
+  return <AstarteDeviceEventDelegate event={event} />;
+};
+
+interface DeviceLiveEventsViewProps {
+  astarte: AstarteClient;
+  deviceId: string;
+}
+
+const DeviceLiveEventsView = ({
+  astarte,
+  deviceId,
+}: DeviceLiveEventsViewProps): React.ReactElement => {
+  const [deviceEvents, setDeviceEvents] = useState<RenderableEvent[]>([]);
+
+  const registerEvent = (event: RenderableEvent) => {
+    setDeviceEvents((oldEvents: RenderableEvent[]) => {
+      const newEvents = oldEvents.slice();
+      newEvents.push(event);
+      return newEvents;
+    });
+  };
+
+  const sendErrorMessage = (errorMessage: string) =>
+    registerEvent({
+      message: errorMessage,
+      level: 'error',
+      timestamp: Date.now(),
+    });
+
+  const sendInfoMessage = (infoMessage: string) =>
+    registerEvent({
+      message: infoMessage,
+      level: 'info',
+      timestamp: Date.now(),
+    });
+
+  useEffect(() => {
+    const onSocketErrorCB = () => sendErrorMessage('Astarte channels communication error');
+    const onSocketCloseCB = () => sendErrorMessage('Lost connection with the Astarte channel');
+
+    astarte.addListener('socketError', onSocketErrorCB);
+    astarte.addListener('socketClose', onSocketCloseCB);
+
+    watchDeviceEvents({
+      astarte,
+      deviceId,
+      onEventReceived: (event: AstarteDeviceEvent) => {
+        registerEvent(event);
+      },
+      onErrorMessage: (msg: string) => {
+        sendErrorMessage(msg);
+      },
+      onInfoMessage: (msg: string) => {
+        sendInfoMessage(msg);
+      },
+    });
+
+    return () => {
+      astarte.removeListener('socketError', onSocketErrorCB);
+      astarte.removeListener('socketClose', onSocketCloseCB);
+      astarte.joinedRooms.forEach((room: string) => {
+        astarte.leaveRoom(room);
+      });
+    };
+  }, [deviceId, astarte]);
+
+  return (
+    <div className="device-event-container p-3">
+      <ul className="list-unstyled">
+        {deviceEvents.map((event, index: number) => (
+          <EventDelegate key={index} event={event} />
+        ))}
+      </ul>
+    </div>
+  );
+};
+
+interface DeviceLiveEventsCardProps {
+  astarte: AstarteClient;
+  deviceId: string;
+}
+
+const DeviceLiveEventsCard = ({
+  astarte,
+  deviceId,
+}: DeviceLiveEventsCardProps): React.ReactElement => {
+  return (
+    <FullHeightCard xs={12} className="mb-4">
+      <Card.Header as="h5">Device Live Events</Card.Header>
+      <Card.Body className="d-flex flex-column">
+        <DeviceLiveEventsView astarte={astarte} deviceId={deviceId} />
```

Comment:
I guess you could now inline `<DeviceLiveEventsView />` since it has exactly the same props as the parent component